File: Easy Challenges/0259 - Clarence the Slow Typist/Solution.java
import java.util.Scanner;
import java.util.stream.IntStream;

public class Solution {

    public static final char[][] KEYPAD = {{'1', '2', '3'}, {'4', '5', '6'}, {'7', '8', '9'}, {'.' ,'0', ' '}};

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.print("Enter an IP address: ");
        System.out.printf("%.2fcm", totalDistance(reader.nextLine().toCharArray()));
    }

    public static double totalDistance(char[] inputs) {
        return IntStream.range(0, arrInputs.length - 1).mapToDouble(i -> distance(keypadCoordinate(arrInputs[i]), keypadCoordinate(arrInputs[i + 1]))).sum();
    }

    public static double distance(int a, int b) {
        return Math.sqrt((Math.pow(a[0] - b[0], 2)) + (Math.pow(a[1] - b[1], 2)));
    }

    public static int[] keypadCoordinate(char c) {
        for (int i = 0; i < KEYPAD.length; i++){
            for (int j = 0; j < KEYPAD[0].length; j++){
                if (KEYPAD[i][j] == c){
                    return new int[]{i, j};
                }
            }
        }
        return null;
    }

}

File: Easy Challenges/0259 - Clarence the Slow Typist/Solution.py
def total_distance(inputs):
    return sum(distance(keyboard_coordinate(i)[1] - keyboard_coordinate(j)[1], keyboard_coordinate(i)[0] - keyboard_coordinate(j)[0]) for i, j in zip(inputs, inputs[1:]))


def distance(a, b):
    return (a ** 2 + b ** 2) ** .5


def keyboard_coordinate(c):
    return divmod('123456789.0'.index(c), 3)


def main():
    print("%.2fcm" % total_distance(input("Enter an IP address: ")))


if __name__ == '__main__':
    main()
